### PR TITLE
MPWG Fee Proposal

### PR DESCRIPTION
**Pull request by starksm64**
Initial proposal for an MPWG fee structure

### CODE REVIEW

File: proposals/working-group/microprofile-working-group/sponsorhsip.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+= Fee Structure Option
+One of the main barriers to a specification process is the cost associated with joining. Companies
+are often willing sponor committee members, spec developers and sign IP commitments, but are unwilling
```
Comment (kwsutter) [inline proposals/working-group/microprofile-working-group/sponsorhsip.adoc]: sponsor (misspelled)

File: proposals/working-group/Comparisons.adoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
+= Pros/Cons of Each Proposal
```
Comment (kwsutter) [inline proposals/working-group/Comparisons.adoc]: Delete this Comparisons.adoc file.  This is the subject of PR #74 and still needs to be fleshed out before posting.

File: proposals/working-group/microprofile-working-group/sponsorhsip.adoc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+to pay a fee to join. We want to support sponorship at two distinct levels; project sponors and specification sponsors.
```
Comment (kwsutter) [inline proposals/working-group/microprofile-working-group/sponsorhsip.adoc]: sponsors (misspelled)

File: proposals/working-group/microprofile-working-group/sponsorhsip.adoc [hunk 4]
```diff
@@ -0,0 +1,31 @@
+= Fee Structure Option
+One of the main barriers to a specification process is the cost associated with joining. Companies
```
Comment (kwsutter) [inline proposals/working-group/microprofile-working-group/sponsorhsip.adoc]: Something's strange with this file...  When I look a rendered version of this .adoc, I get the title (Fee Structure Option), but then the text starts with line 4 (to pay a fee...).

**Comment by kwsutter** [comment]
@starksm64 Just merged, but then I saw that the name of the file is misspelled...  sponsorhsip.adoc   :-(  And, I would still provide a link to this from the MPWG doc or else it's just kind of lost...  Nobody can easily find it.  Thanks!

**Comment by gunnarmorling** [comment]
Came here to point out the typo in the filename, too. But also a question: can an org be both at once, project sponsor and spec sponsor? Would it do so in order to have influence via spec work and reputation from the project sponsorship via prominent website placement? Might be worth adding a sentence that one can have both roles, if true.

**Comment by starksm64** [comment]
> But also a question: can an org be both at once, project sponsor and spec sponsor? Would it do so in order to have influence via spec work and reputation from the project sponsorship via prominent website placement? Might be worth adding a sentence that one can have both roles, if true.

Both project and spec sponsors can fund spec developers. A spec sponsor is just a way for a company to join the MP working group and sign the working group agreement for IP capture without having to pay a separate fee. I can clarify that project sponsors are generally doing both.

**Comment by starksm64** [comment]
The filename and Gunnar's questions about roles has been addressed in #77

**Comment by waynebeaton** [comment]
Working groups are a means for organizations to collaborate. Open source projects are means for developers to collaborate. Working groups do not _own_ open source projects. With the exception of the specification committee's authority and responsibility regarding reviews for specification projects, working groups have no special authority over any open source projects that they consider to be under their purview. So, there is no direct line of authority (the role of the specification committee notwithstanding) from a working group to any open source project that permits the working group to instruct an open source project to do anything, including listing sponsors.

For completeness, a working group influences the direction of an open source project by having committers representing the interests of working group participants.

There are two ways to put a company logo on an Eclipse open source project (specification projects included): as a entry in a list of adopters without preference or bias, or as a project contributor. In the latter case, a company name and logo may be listed on a project page when the company is a member of the Eclipse Foundation and that company has at least one committer that has contributed at least one commit in the last three months. This is captured in the [Eclipse Foundation Project Handbook](https://www.eclipse.org/projects/handbook/#company-logos).

The notion of direct/overt project sponsorship goes against vendor neutrality. A company may show their support for an open source project by contributing development resources.

Working groups are, however, not open source projects, and you have considerably more leeway. It is completely reasonable for a working group's website to provide preferential display of company logos to members, perhaps based on participation level.

**Comment by starksm64** [comment]
The proposed indication of sponsorship levels is targeting the working group page, not specification project pages.

**Comment by waynebeaton** [comment]
> The proposed indication of sponsorship levels is targeting the working group page, not specification project pages.

That is not clear to me from the text. "Project" has specific meaning in the Eclipse Foundation Development Process.

**Pull request merged**